subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/DefaultPropertyMetadataStore.java

```diff
@@ -142,6 +149,7 @@ public DefaultPropertyMetadataStore(Iterable<? extends PropertyAnnotationHandler
         return ImmutableSet.<Class<? extends Annotation>>builder()
             .addAll(propertyTypeAnnotations)
             .add(Optional.class)
+            .add(Nullable.class)
```

Comment:
Do we need to do all this gymnastics here? I don't think we need to process `@Nullable` here.

Reply:
If we don't add `@Nullable` here, then the `@Nullable` annotation is not copied to `DefaultPropertyMetadata`, so the check if it is present would always be false.
IOW, leaving out `@Nullable` from this list has not the desired effect.

Reply:
Makes sense!